__tests__/app.test.js

```diff
@@ -27,174 +27,280 @@ describe("GET /api/topics", () => {
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+  test("200: returns array of article objects arranged by date in ascending order", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy("created_at", { descending: false });
+      });
+  });
+  test('200: returns array of article objects without a "body" property', () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles.length).toBe(13);
+        articles.forEach((article) => {
+          expect(article).not.toHaveProperty("body");
+        });
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: responds with success code when comment is created, returns object of posted comment", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best article ever",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.postedComment).toEqual({
+          article_id: 2,
+          author: "butter_bridge",
+          body: "best article ever",
+          comment_id: expect.any(Number),
           created_at: expect.any(String),
-          votes: expect.any(Number),
-          article_img_url: expect.any(String),
-          comment_count: expect.any(String)
+          votes: 0,
         });
+      });
+  });
+  test("404: responds error message of 'not found' if article doesn't exist", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best article ever",
+    };
+    return request(app)
+      .post("/api/articles/100/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("400: responds error message of 'bad request' if article_id is wrong type", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "best article ever",
+    };
+    return request(app)
+      .post("/api/articles/cheese/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+  test("400: responds error message of 'user does not exist' if user doesn't exist", () => {
+    const newComment = {
+      username: "vh232",
+      body: "best article ever",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("user does not exist");
+      });
+  });
```

Comment:
Should this be a 400 status code? if the user vh232 does not exist does that make them an invalid user or a user that does not exist?